### ISSUE TITLE
Volume disappeared after rebooting, second attempt pre-boot

### ISSUE THREAD

**Issue by Neobond** [007revad/Synology_HDD_db#373]
I am running DSM 7.2.2 via ARC Patch on a F4-424 Max with a SHR volume on 2x NVMe and another volume on 2x HDDs. The last time I ran this script, DSM wouldn't boot, I think the volume on the NVMe was removed, even though I saved and ran the script from the HDD (volume2).

Just wondering if this looks good (have not rebooted yet, so I still have the firmware error for both TEAMGROUP SSDs)

![2024-10-22_095847](https://github.com/user-attachments/assets/c68ad938-57e9-4097-85c8-609b65154684)

And I created a boot task to run on boot (as root).

![2024-10-22_100345](https://github.com/user-attachments/assets/027755a3-e3a8-4f26-89d9-7204cc9212a0)

Kind of worried to reboot :P

**Comment by 007revad** [comment]
I see 2 issues: 
1. The script determined that the OOS20000G is 20000 GB but the sa6400_host_v7.db is showing 19999 GB
2. TEAM TM8FFD004T does not look correct. Googling for "team tm8ffd004t" or "tm8ffd004t" finds nothing. 

To fix issue 1, run the script once with the --restore option to undue the previous changes, then run the script as normal.

For issue 2, what does "Storage Manager > HDD/SSD" show as the Model and Firmware? 
![image](https://github.com/user-attachments/assets/0d8bcd9c-d38a-4356-9159-19a836939ccc)

**Comment by Neobond** [comment]
Thanks for the reply. The 20TB drives are whitelabel (Seagate X22 20TB)

I followed the instructions to restore and reran the script, and it seems to add back the same 20000 data for the HDDs

![2024-10-22_110606](https://github.com/user-attachments/assets/9f8b34c0-dea8-45d0-8ae1-243b0268ce0c)

Here is a snapshot of the drives in Storage Manager, they are TEAMGROUP MP44Q 4TB drives sent directly from the manufacturer (they are ex review drives) I have 4 in total and the other two do show T-FORCE in the name.

![2024-10-22_110716](https://github.com/user-attachments/assets/ca1787cc-4752-4d91-bdb2-1237b638effe)

what the 4 drives look like
![20240910_155921](https://github.com/user-attachments/assets/d81b52bb-3bdd-4e09-a9f2-8a7e00da0db2)

Other two showing T-FORCE in the name
![2024-10-06_201847](https://github.com/user-attachments/assets/96cd729c-660d-415a-a4e5-b31b92401ae3)

But they are stickered the same!?

**Comment by Neobond** [comment]
@007revad any idea if I can proceed safely with the screenshots I provided?

I was also wondering, if I disabled the compatibility list from updating, I wouldn;'t need to run the script on every boot right?

**Comment by 007revad** [comment]
Even with the compatible drives list updates disabled DSM will still update the list when you do a DSM update.

- ARC addons hdddb addon is v3.5.101 of this syno_hdd_db script.
- TCRP addons hdddb addon is v3.5.102 of this syno_hdd_db script.

Both of them run the script during the junior boot process, and their loaders edit the model.dtb file (if it's a device tree model).

So if you have the hdddb addon enabled with different options it could  undo what the syno_hdd_db script did when you ran it.

**Comment by Neobond** [comment]
Ahh thanks, yes I do have that addon enabled with ARC bootloader, so I would have to go to the bootloader and disable the ARC addon compatibility, rebuild the loader ands boot to DSM, which should then just trigger yours, and fix the firmware update message? 

I am more worried if my SSD volume will disappear when using the script.

**Comment by 007revad** [comment]
Your SSD volume should not disappear, but I can't be 100% sure because I don't know what caused it to disappear.

**Comment by Neobond** [comment]
I installed Arc Control and the disk and memory compatibility are unticked by default, I think it is because it was applied once in the original boot loader (to allow installation) but I can't be sure.

![2024-10-24_171550](https://github.com/user-attachments/assets/b27ceecd-f6cf-402a-9376-2b32ae54e882)

I am going to reboot to the loader and check the addons, if I lose the SSD volume I will have to do some more testing before I can start migrating.

**Comment by 007revad** [comment]
Is Arc Control a package? I assume it is included with Arc Loader?

I'd like to be able to have a look at Arc Control without having to setup a XPE box.

**Comment by Neobond** [comment]
Can you DM me or something because the developer does not allow sharing of ARC files.

Anyway I disabled the Arc compatibility addon/unticked synoboot compatibilty checking, and rebooted with just the script active and I still get the unrecognized firmware message.

**Comment by Neobond** [comment]
I am not sure what got it working, possibly updating the ARC loder boot image from 24.10.04 to the latest one (24.10.25) I reenabled the hdddb addon, also ensured HDD and memory compatibility was checked in Synoboot options, and now the unknown firmware message is gone.

![2024-10-26_161312](https://github.com/user-attachments/assets/13e9ef70-d435-417e-9579-a8cddaee17e3)
![2024-10-26_161330](https://github.com/user-attachments/assets/99abdcc0-1de4-42a5-a130-090584a01102)

I am pretty sure I started off with this same exact config, so I am just going to blame it on the bootloader version difference. 

Thanks for your help!
